pkg/database/database.go

```diff
@@ -102,6 +148,66 @@ func (database *DatabaseHandler) GetVirtualExpireIndexes(port uint64) (map[strin
 	return virtualIndex, expireIndex, nil
 }
 
+func (database *DatabaseHandler) AddToExpireIndex(port uint64, dbname string, filename string, lsn uint64) error {
+
+	databases, err := getDatabase(port)
+	if err != nil || databases == nil {
+		return fmt.Errorf("unable to get ao/aocs tables %v", err) // fix
+	}
+	for _, db := range databases {
+		if dbname != db.name {
+			continue
+		}
+		ylogger.Zero.Debug().Str("database name", db.name).Msg("received database")
+		conn, err := connectToDatabase(port, db.name)
```

Comment:
This is extremely slow to get database connections every time. We need to open connection once, then pass it to this function